utils/index.sh

```
@@ -73,6 +73,100 @@ setup(){
     export ES_INDEX=$ES_INDEX
+
+    if [[ "${PLATFORM:-openshift}" == "microshift" ]]; then
+        # MicroShift does not expose many OpenShift config APIs. Keep this
+        # metadata path limited to Kubernetes resources and the
+        # microshift-version ConfigMap.
+        cluster_name=${MICROSHIFT_CLUSTER_NAME:-$(kubectl get ns kube-system -o jsonpath='{.metadata.uid}' 2>/dev/null || hostname -s 2>/dev/null || echo "microshift")}
+        # Prefer the microshift-version CM (carries the 4.y release); fall
+        # back to oc version (Kubernetes gitVersion) only if the CM is absent.
+        if mscm=$(kubectl get cm -n kube-public microshift-version -o json 2>/dev/null); then
+            cluster_version=$(echo "$mscm" | jq -r '.data.version // "unknown"')
+            export RELEASE_STREAM=$(echo "$mscm" | jq -r '"\(.data.major).\(.data.minor)"')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)